sphinx_gallery/directives.py

```diff
@@ -0,0 +1,79 @@
+"""
+Custom Sphinx directives
+========================
+"""
+
+import os
+
+from docutils import statemachine
+from docutils.parsers.rst import Directive, directives
+
+
+class MiniGallery(Directive):
+    """
+    Custom directive to insert a mini-gallery
+
+    The required argument is one or more fully qualified names of objects,
+    separated by spaces.  The mini-gallery will be the subset of gallery
+    examples that make use of that object (from that specific namespace).
+
+    Options:
+
+    * `add-heading` adds a heading to the mini-gallery.  If an argument is
+      provided, it uses that text for the heading.  Otherwise, it uses
+      default text.
+    * `heading-level` specifies the heading level of the heading as a single
+      character.  If omitted, the default heading level is `'^'`.
+    """
+    required_arguments = 1
+    optional_arguments = 0
+    final_argument_whitespace = True
+    option_spec = {'add-heading': directives.unchanged,
+                   'heading-level': directives.single_char_or_unicode}
+
+    def run(self):
+        # Respect the same disabling options as the `raw` directive
+        if (not self.state.document.settings.raw_enabled
+                or not self.state.document.settings.file_insertion_enabled):
+            raise self.warning('"%s" directive disabled.' % self.name)
```

Comment:
Sorry, just a quick question, if the directive is disabled, does the rest of the code continue to run or does the warning stop it? @ayshih

Reply:
Sorry for the slow response.  The rest of the code should not continue to run if the directive is disabled.

Reply:
Thank you!